pkg/sdk/testint/functions_integration_test.go

```diff
@@ -0,0 +1,407 @@
+package testint
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_CreateFunctions(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	_, warehouseCleanup := createWarehouse(t, client)
+	t.Cleanup(warehouseCleanup)
+	databaseTest, databaseCleanup := createDatabase(t, client)
+	t.Cleanup(databaseCleanup)
+	schemaTest, schemaCleanup := createSchema(t, client, databaseTest)
+	t.Cleanup(schemaCleanup)
+
+	cleanupFunctionHandle := func(id sdk.SchemaObjectIdentifier, argumentTypes []sdk.DataType) func() {
+		return func() {
+			es := []sdk.FunctionArgumentTypeRequest{}
+			for _, item := range argumentTypes {
+				es = append(es, *sdk.NewFunctionArgumentTypeRequest().WithArgDataType(item))
+			}
+			err := client.Functions.Drop(ctx, sdk.NewDropFunctionRequest(id).WithArgumentTypes(es))
+			if errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
+				return
+			}
+			require.NoError(t, err)
+		}
+	}
+
+	t.Run("create function for Java", func(t *testing.T) {
+		name := "echo_varchar"
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		definition := `
+	class TestFunc {
+		public static String echoVarchar(String x) {
+			return x;
+		}
+	}`
+		target := fmt.Sprintf("@~/tf-%d.jar", time.Now().Unix())
+		returnsRequest := sdk.NewFunctionReturnsRequest().WithResultDataType(sdk.DataTypeVARCHAR)
+		argumentRequest := sdk.NewFunctionArgumentRequest().WithArgName("x").WithArgDataType(sdk.DataTypeVARCHAR)
+		request := sdk.NewCreateFunctionForJavaFunctionRequest(id).
+			WithOrReplace(sdk.Bool(true)).
+			WithArguments([]sdk.FunctionArgumentRequest{*argumentRequest}).
+			WithNullInputBehavior(sdk.FunctionNullInputBehaviorCalledOnNullInput).
+			WithReturns(returnsRequest).
+			WithHandler(sdk.String("TestFunc.echoVarchar")).
+			WithTargetPath(&target).
+			WithFunctionDefinition(&definition)
+		err := client.Functions.CreateFunctionForJava(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupFunctionHandle(id, []sdk.DataType{"VARCHAR"}))
+
+		functions, err := client.Functions.Show(ctx, sdk.NewShowFunctionRequest())
+		require.NoError(t, err)
+		require.Equal(t, 1, len(functions))
+	})
+
```

Comment:
There's no test for Java jar (stage) scenario

Reply:
i would rather not write a test for stage at this time, it would be fairly complex. same as for procedures.

Reply:
True, leave a comment or create a follow up task for it

Reply:
added comment

> // todo: add tests for:
> //  - creating functions with different languages from stages  i.e. [ TARGET_PATH = '<stage_path_and_file_name_to_write>' ]
> //  - execute and execute-immediate for scripting https://docs.snowflake.com/en/sql-reference/sql/execute-immediate

Reply:
added comment
/*
todo: add tests for:
  - creating functions with different languages (java, javascript, python, scala, sql) from stages  using [ TARGET_PATH = '<stage_path_and_file_name_to_write>' ]
  - execute and execute-immediate for scripting https://docs.snowflake.com/en/sql-reference/sql/execute-immediate
*/